identity/src/main/java/com/android/identity/credential/CredentialStore.java

```diff
@@ -77,6 +77,28 @@ public CredentialStore(@NonNull StorageEngine storageEngine,
                 credentialKeySettings);
     }
 
+    /**
+     * Creates a new credential using a key which already exists in some keystore.
+     *
+     * <p>If a credential with the given name already exists, it will be overwritten by the
+     * newly created credential.
+     *
+     * @param name name of the credential.
+     * @param credentialKeySettings the settings to use for CredentialKey.
+     * @param existingKeyAlias the alias of the existing key.
+     * @return A newly created credential.
+     */
+    public @NonNull Credential createCredentialWithExistingKey(
```

Comment:
I would prefer for this to not be public as it's only called by `migrateToCredentialStore()`. Maybe add "@hide" for now and we can revisit when solving #292.